### PR TITLE
Enable theme switching and inline themes

### LINKED ISSUE #120: As a user I need to override the default tokens with a simple configuration
Examples of simple configuration:
- [ ] `polaris.config.js` file, similar to [tailwindcss](https://tailwindcss.com/docs/configuration#creating-your-configuration-file)

### PR DESCRIPTION
**Pull request by samrose3**
This is an experiment to see how themes are used and customized. This PR tests:

- creating a custom theme from the base theme using `createTheme` in a `.css.ts` file (`vanilla-extract` required)
- creating an **inline theme** using `createInlineTheme` in a normal `.ts` file (**no** `vanilla-extract` required)

The current theme overrides are being defined in `polaris.config.js`, a pattern that is similar to Tailwind's `tailwind.config.js`. Once fully configured, this would make theme overrides standardized and easy to apply.

Still to-be-done here:

- load the `polaris.config.js` using a [find-up](https://github.com/sindresorhus/find-up) strategy, then use those theme token values instead of the defaults

theme-provider

Closes #120

### CODE REVIEW

**Comment by samrose3** [comment]
@alex-page just a follow up on the desired `polaris.config.json` strategy for overriding.

This could still be possible, but it would require tapping in to the consumers build setup (Rollup, webpack, esbuild). We can't import/require to the config during runtime of the ThemeProvider.

I fiddled with trying load overrides even in a `.css.ts` file which is processed at build time; however, it would require a top-level `await` or Promise since the file path to the config would need to be a dynamic import (ex: `await import(polarisConfigPath)`). Vite was converting `required(polarisConfigPath)` to imports unless the build type was changed but that still didn't work.

In summary, I'm a bit stumped on this `.config` 😓

**Comment by alex-page** [comment]
@samrose3 this is an awesome investigation. Lets pause this for now but I really appreciate all the context and learnings you have gathered.

**Comment by alex-page** [comment]
Lets leave it here for now, as we don't need this functionality just yet.

**Comment by samrose3** [comment]
Going to close this PR since it is outta date and uncertain if we want this functionality now.

We can reference it for future theming needs.
